### DISCUSSION TITLE
link -lib) interface... unknown configure: error: could not determine link -lib interface

### DISCUSSION THREAD

**Discussion by xzatech** [lldpd/lldpd#545] [Q&A]
.../lldpd-1.0.15/build $ ../configure
checking for a BSD-compatible install... /data/data/com.termux/files/usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /data/data/com.termux/files/usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '10568' is supported by ustar format... yes
checking whether GID '10568' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for ar... no
checking for lib... no
checking for link... link -lib
checking the archiver (link -lib) interface... unknown
configure: error: could not determine link -lib interface

**Comment by ShumaDK** [comment]
`pkg install binutils` helped me in Termux

[davy.ai](https://davy.ai/author/davy-ai/)
[October 16, 2023 at 1:11 pm](https://davy.ai/termux-autotools-libtool-cannot-determine-link-lib-interface/#comment-1325474)

The error message “checking the archiver (link -lib) interface… unknown configure: error: could not determine link -lib interface” indicates that the configure script is unable to determine the correct linker interface for the archiver.

To resolve this issue, you can try the following steps:

Make sure you have the necessary build tools and dependencies installed on your Termux environment. Check if you have the libtool package installed by running the command pkg list-installed | grep libtool. If it is not installed, you can install it by running pkg install libtool.
Make sure the libtool package is up-to-date by running pkg update libtool.

Check if the archiver is available on your system. You can run the command ar --version to check if it is installed. If it is not installed, you can install it by running pkg install binutils.

If the archiver is installed, but the issue persists, you may need to specify the linker interface manually in the configure script. To do this, open the configure script in a text editor and search for the line that checks the archiver. It might look something like this:

AC_CHECK_PROG(LINK, $AR -lib, yes, no)
Replace it with the following line:

AC_CHECK_PROG(LINK, $AR rc, yes, no)
This change tells the configure script to use the “rc” command for the archiver rather than “-lib”.

Save the modified configure script and run autoreconf -i and ./configure again. The issue should be resolved now.

If the issue still persists, there might be some other underlying problems with your setup or project configuration. In that case, you may need to investigate further or seek assistance from the project’s developers or community.
